tests/common/connectors/related-products/options.ts

```diff
@@ -0,0 +1,232 @@
+        await setup(options);
+      }).rejects.toThrowErrorMatchingInlineSnapshot(`
+              "The \`objectID\` option is required.
+
+              See documentation: https://www.algolia.com/doc/api-reference/widgets/related-products/js/#connector"
+            `);
+    });
+
+    test('forwards parameters to the client', async () => {
+      const searchClient = createMockedSearchClient();
+      const options: SetupOptions<RelatedProductsConnectorSetup> = {
+        instantSearchOptions: { indexName: 'indexName', searchClient },
+        widgetParams: {
+          objectID: '1',
+          maxRecommendations: 2,
+          threshold: 3,
+          fallbackParameters: { facetFilters: ['test1'] },
+          queryParameters: { analytics: true },
+        },
+      };
+
+      await setup(options);
+
+      expect(searchClient.getRecommendations).toHaveBeenCalledWith(
+        expect.arrayContaining([expect.objectContaining(options.widgetParams)])
+      );
+    });
+
+    test('returns recommendations', async () => {
+      const options: SetupOptions<RelatedProductsConnectorSetup> = {
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient: createMockedSearchClient(),
+        },
+        widgetParams: { objectID: '1' },
+      };
+
+      await setup(options);
+
+      expect(screen.getByRole('list')).toMatchInlineSnapshot(`<ul />`);
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(screen.getByRole('list')).toMatchInlineSnapshot(`
+        <ul>
+          <li>
+            A0E200000002BLK
+          </li>
+          <li>
+            A0E200000001WFI
+          </li>
+          <li>
+            A0E2000000024R1
+          </li>
+        </ul>
+      `);
+    });
+
+    test('transforms recommendations', async () => {
+      const options: SetupOptions<RelatedProductsConnectorSetup> = {
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient: createMockedSearchClient(),
+        },
+        widgetParams: {
+          objectID: '1',
+          transformItems(items) {
+            return items.map((item) => ({
+              ...item,
+              objectID: item.objectID.toLowerCase(),
+            }));
+          },
+        },
+      };
+
+      await setup(options);
+
+      expect(screen.getByRole('list')).toMatchInlineSnapshot(`<ul />`);
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(screen.getByRole('list')).toMatchInlineSnapshot(`
+        <ul>
+          <li>
+            a0e200000002blk
+          </li>
+          <li>
+            a0e200000001wfi
+          </li>
+          <li>
+            a0e2000000024r1
+          </li>
+        </ul>
+      `);
+    });
+  });
+}
+
+function createMockedSearchClient() {
+  return createSearchClient({
+    getRecommendations: jest.fn((requests) =>
+      Promise.resolve(
+        createMultiSearchResponse(
+          ...requests.map((request) => {
```

Comment:
I had the same issue. Basically in algoliasearch v3 `getRecommendations` doesn't exist so everything in it is not typed. We can safely ignore this.

```suggestion
          // @ts-ignore
          // `request` will be implicitly typed as any in type-check:v3
          // since `getRecommendations` is not available there
          ...requests.map((request) => {
```